### PR TITLE
Query notifications

### PR DESCRIPTION
**Pull request by cybrox**
Refactor notifications to use query parameters.

Considering we duplicated the routine for marking notifications as read/marked here, it might be worth moving that into a utility as well.

**Please Note:** This still requires minor adjustments to be compatible with the latest changes in notifications and probably some syntax issues need fixing. Will look into that asap.

cc @vevix

### CODE REVIEW

File: app/controllers/application.js
Comment (trmcnvn) [inline app/controllers/application.js]: no need to `return` in an observer

**Review by trmcnvn — DISMISSED** [review]
:+1: Looks good, need to fix conflicts and eslint error.

**Comment by cybrox** [comment]
Should be ready to squash+merge, once Travis has finished.

**Comment by cybrox** [comment]
Should be ready to squash+merge, once Travis has finished.
I forgot how to git for a second.

File: app/components/application/user-notifications/item.js
Comment (trmcnvn) [inline app/components/application/user-notifications/item.js]: Needs updating.

**Comment by cybrox** [comment]
Technically, this is cleaned up and ready for merge.
Might want to hold off until @NuckChorris fixed the feed issues, so I can test it again locally. I guess we want to make super-sure nothing more breaks right now.

**Comment by trmcnvn** [comment]
@cybrox I will push this to staging

**Comment by cybrox** [comment]
Looks like it's working. I should probably add some code to mark the notification as read client-side as well, if it loaded (=visible in the notification dropdown)

**Comment by trmcnvn** [comment]
Yep, good catch.

File: app/controllers/application.js
Comment (trmcnvn) [inline app/controllers/application.js]: `filterBy('streamId', notification)`

File: app/controllers/application.js
Comment (trmcnvn) [inline app/controllers/application.js:22]: We should check that `session.hasUser()` is true just in case here.

File: app/controllers/application.js
Comment (trmcnvn) [inline app/controllers/application.js]: `const [trackedItem] = trackedItems;`

**Review by trmcnvn — DISMISSED** [review]
:+1: Will merge after final cleanup!

**Pull request merged**
